src/IrcConnection/IrcConnection.cs

```diff
@@ -1188,16 +810,27 @@ private class WriteThread
             private int            _BelowMediumThresholdCount = 1;
             private int            _BurstCount;
 
-            public AutoResetEvent QueuedEvent;
+            private AutoResetEvent _QueuedEvent;
```

Comment:
This should have been a auto-property but became a public field. In private inner-classes that isn't very harmful (nobody can access it except the outer class), so use the original name and add { get; set; } here